integ-test/src/test/java/org/opensearch/sql/sql/NestedIT.java

```diff
@@ -0,0 +1,116 @@
+    loadIndex(Index.NESTED_WITHOUT_ARRAYS);
+  }
+
+  @Test
+  public void nested_function_with_array_of_nested_field_test() {
+    String query = "SELECT nested(message.info), nested(comment.data) FROM " + TEST_INDEX_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(6, result.getInt("total"));
+    verifyDataRows(result,
+        rows("c", "ab"),
+        rows("a", "ab"),
+        rows("b", "aa"),
+        rows("c", "aa"),
+        rows("a", "ab"),
+        rows("zz", new JSONArray(List.of("aa", "bb"))));
+  }
+
+  @Test
+  public void nested_function_in_select_test() {
+    String query = "SELECT nested(message.info), nested(comment.data), "
+        + "nested(message.dayOfWeek) FROM "
+        + TEST_INDEX_NESTED_TYPE_WITHOUT_ARRAYS;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(5, result.getInt("total"));
+    verifySchema(result,
+        schema("message.info", null, "keyword"),
+        schema("comment.data", null, "keyword"),
+        schema("message.dayOfWeek", null, "long"));
+    verifyDataRows(result,
+        rows("a", "ab", 1),
+        rows("b", "aa", 2),
+        rows("c", "aa", 1),
+        rows("c", "ab", 4),
+        rows("zz", "bb", 6));
+  }
+
+  // Has to be tested with JSON format when https://github.com/opensearch-project/sql/issues/1317
+  // gets resolved
+  @Disabled // TODO fix me when aggregation is supported
+  public void nested_function_in_an_aggregate_function_in_select_test() {
+    String query = "SELECT sum(nested(message.dayOfWeek)) FROM " +
+        TEST_INDEX_NESTED_TYPE_WITHOUT_ARRAYS;
+    JSONObject result = executeJdbcRequest(query);
+    verifyDataRows(result, rows(14));
+  }
+
+  // TODO Enable me when nested aggregation is supported
+  @Disabled
+  public void nested_function_with_arrays_in_an_aggregate_function_in_select_test() {
+    String query = "SELECT sum(nested(message.dayOfWeek)) FROM " +
+        TEST_INDEX_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+    verifyDataRows(result, rows(19));
+  }
+
+  // TODO not currently supported by legacy, should we add implementation in AstBuilder?
+  @Disabled
+  public void nested_function_in_a_function_in_select_test() {
+    String query = "SELECT upper(nested(message.info)) FROM " +
+        TEST_INDEX_NESTED_TYPE_WITHOUT_ARRAYS;
+    JSONObject result = executeJdbcRequest(query);
+
+    verifyDataRows(result,
+        rows("A"),
+        rows("B"),
+        rows("C"),
+        rows("C"),
+        rows("ZZ"));
+  }
+
+
+  @Test
+  public void nested_function_with_array_of_multi_nested_field_test() {
+    String query = "SELECT nested(message.author.name) FROM " + TEST_INDEX_MULTI_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(6, result.getInt("total"));
+    verifyDataRows(result,
+        rows("e"),
+        rows("f"),
+        rows("g"),
+        rows("h"),
+        rows("p"),
+        rows("yy"));
+  }
```

Comment:
Add a test with `nested` with condition
Add test(s) with incorrect input (missing subfield or path is not of `nested` type)

Reply:
> Nested with date type subfields are supported however it returns the wrong type. Maybe date types need to be included in NestedFunction.java. **Legacy:** <img alt="Screenshot 2023-03-20 at 3 32 13 PM" width="515" src="https://user-images.githubusercontent.com/29149268/226479813-81462e6d-3860-43c0-be73-b4767106251d.png"> **V2:** <img alt="Screenshot 2023-03-20 at 3 32 28 PM" width="495" src="https://user-images.githubusercontent.com/29149268/226479862-d2655a38-45e3-4827-a7a2-371093a51502.png">

Appears to be a bug with not only nested types!

Reply:
Agree, but please add tests 
1. with condition
2. with null and missing cases
For example, index:
```
doc_1: "message": { "info" : "something" }
doc_2: "message": { "info" : "another thing" }
doc_3: "message": { "info" : null }
doc_4: "message": { }
doc_5: // no message
```
Maybe combine this with multiple `nested` calls and/or deeper nesting to prove that code in `UnnestOperator` is solid.